Fix ARM and ARM64 bootstrap on modern Debian sid

pulkomandy opened the pull request:
texinfo 4 fails to build on arm64, there is a mixup of size_t and int type passing arguments to ansi_escape. (32 bit platforms won't have a problem here, but 64 bit ones will)

I attempted to use a newer version of texinfo, however this fails while cross-compiling the internal gnulib:

```
libtool: compile:  x86_64-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I./parsetexi -I. -I./gnulib/lib -I./gnulib/lib -DDATADIR=\"/packages/texinfo_bootstrap-7.1-1/.self/data\" -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -DVERSION=\"0\" -DXS_VERSION=\"0\" -I/usr/lib/x86_64-linux-gnu/perl/5.40/CORE -MT parsetexi/Parsetexi_la-source_marks.lo -MD -MP -MF parsetexi/.deps/Parsetexi_la-source_marks.Tpo -c parsetexi/source_marks.c  -DPIC -o parsetexi/.libs/Parsetexi_la-source_marks.o
libtool: link: x86_64-linux-gnu-gcc -shared  .libs/XSParagraph.o .libs/xspara.o .libs/text.o  -Wl,--whole-archive ./gnulib/lib/.libs/libgnu.a -Wl,--no-whole-archive   -O2 -g -Wl,-E   -Wl,-soname -Wl,XSParagraph.so -o .libs/XSParagraph.so
/usr/bin/ld: ./gnulib/lib/.libs/libgnu.a(libgnu_la-obstack.o): warning: relocation against `obstack_alloc_failed_handler' in read-only section `.text'
libtool: link: x86_64-linux-gnu-gcc -shared  .libs/MiscXS.o .libs/misc.o  -Wl,--whole-archive ./gnulib/lib/.libs/libgnu.a -Wl,--no-whole-archive   -O2 -g -Wl,-E   -Wl,-soname -Wl,MiscXS.so -o .libs/MiscXS.so
/usr/bin/ld: .libs/xspara.o: relocation R_X86_64_PC32 against symbol `stderr@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[5]: *** [Makefile:1615: XSParagraph.la] Error 1
make[5]: *** Waiting for unfinished jobs....
/usr/bin/ld: ./gnulib/lib/.libs/libgnu.a(libgnu_la-obstack.o): warning: relocation against `obstack_alloc_failed_handler' in read-only section `.text'
/usr/bin/ld: ./gnulib/lib/.libs/libgnu.a(libgnu_la-obstack.o): relocation R_X86_64_PC32 against symbol `obstack_alloc_failed_handler' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[5]: *** [Makefile:1548: MiscXS.la] Error 1
mv -f parsetexi/.deps/Parsetexi_la-source_marks.Tpo parsetexi/.deps/Parsetexi_la-source_marks.Plo
make[5]: Leaving directory '/home/pulkomandy/haiku/generated.arm64/objects/haiku/arm64/packaging/repositories/HaikuPortsCross-build/sys-apps/texinfo_bootstrap/work-7.1/sources/texinfo-7.1/tp/Texinfo/XS'
make[4]: *** [Makefile:1908: all-recursive] Error 1
make[4]: Leaving directory '/home/pulkomandy/haiku/generated.arm64/objects/haiku/arm64/packaging/repositories/HaikuPortsCross-build/sys-apps/texinfo_bootstrap/work-7.1/sources/texinfo-7.1/tp/Texinfo/XS'
make[3]: *** [Makefile:1459: all] Error 2
make[3]: Leaving directory '/home/pulkomandy/haiku/generated.arm64/objects/haiku/arm64/packaging/repositories/HaikuPortsCross-build/sys-apps/texinfo_bootstrap/work-7.1/sources/texinfo-7.1/tp/Texinfo/XS'
make[2]: *** [Makefile:4351: all-recursive] Error 1
make[2]: Leaving directory '/home/pulkomandy/haiku/generated.arm64/objects/haiku/arm64/packaging/repositories/HaikuPortsCross-build/sys-apps/texinfo_bootstrap/work-7.1/sources/texinfo-7.1/tp'
make[1]: *** [Makefile:1769: all-recursive] Error 1
make[1]: Leaving directory '/home/pulkomandy/haiku/generated.arm64/objects/haiku/arm64/packaging/repositories/HaikuPortsCross-build/sys-apps/texinfo_bootstrap/work-7.1/sources/texinfo-7.1'
make: *** [Makefile:1708: all] Error 2
Command '['bash', '-c', '. /home/pulkomandy/haiku/generated.arm64/objects/haiku/arm64/packaging/repositories/HaikuPortsCross-build/sys-apps/texinfo_bootstrap/work-7.1/wrapper-script']' returned non-zero exit status 2.
```

I had to remove the libtoolize call from the build because that fails in another way (I assume the version of libtool is too old?)

What's the best way to go about this? The patch for texinfo4 is trivial, but I'd rather not stay forever on an old version of it?

pulkomandy committed:
texinfo: reorder fields to match the non-bootstrap recipe

pulkomandy committed:
Import textinfo 7.1 to try to fix arm64 bootstrap

jmairboeck commented:
Isn't texinfo 4 only needed for gcc2? Everything else should be able to use texinfo 7.

pulkomandy commented:
Yes, but

1) currently this does not work
2) I am generally careful to not change other architectures in haikuports.cross without testing the build first.

I'll test them and update as needed.

pulkomandy committed:
Update Texinfo to 7.2

- Use cross compiling patches from Yocto with some additional hacks
- Use out of source build, building inside the source dir doesn't work
- Force -fPIC in compiler flags
- Disable perl modules we are not interested in

This results in a succesfully built package on a current version of
Debian Sid. Hopefully it works on other setups as well.

pulkomandy commented:
I tried again with texinfo 7.2 and some additional patches from Yocto and managed to get it building on my setup.
I will do a forum post asking for testing and feedback on this.

pulkomandy committed:
Switch arm 32 to texinfo 7.3 as well

pulkomandy committed:
bash: move arm 32 to bash 5.3, remove obsolete recipes
